dev-tools/mage/crossbuild.go

```diff
@@ -315,18 +317,26 @@ func (b GolangCrossBuilder) Build() error {
 
 	if runtime.GOOS != "windows" {
 		args = append(args,
-			"--env", "EXEC_UID="+strconv.Itoa(os.Getuid()),
-			"--env", "EXEC_GID="+strconv.Itoa(os.Getgid()),
+			"--env", fmt.Sprintf("EXEC_UID=%d", uid),
+			"--env", fmt.Sprintf("EXEC_GID=%d", gid),
 		)
 	}
 	if versionQualified {
 		args = append(args, "--env", "VERSION_QUALIFIER="+versionQualifier)
 	}
-	if CrossBuildMountModcache {
-		// Mount $GOPATH/pkg/mod into the container, read-only.
-		hostDir := filepath.Join(build.Default.GOPATH, "pkg", "mod")
-		args = append(args, "-v", hostDir+":/go/pkg/mod:ro")
+
+	// Mount $GOPATH/pkg/mod into the container, read-only.
+	hostDir := filepath.Join(build.Default.GOPATH, "pkg", "mod")
+	args = append(args, "-v", fmt.Sprintf("%s:/go/pkg/mod:ro", hostDir))
+
+	// Mount the go build cache into the container.
+	out, err := exec.Command("go", "env", "GOCACHE").Output()
+	if err != nil {
+		return fmt.Errorf("failed to get GOCACHE: %w", err)
 	}
+	cacheDir := strings.TrimSpace(string(out))
+	buildCacheMountPoint := "/tmp/.cache/go-build"
+	args = append(args, "-v", fmt.Sprintf("%s:%s", cacheDir, buildCacheMountPoint))
```

Comment:
I would strongly recommend an escape hatch here, so if this ever creates issues we can easily disable it. E.g. you could easily have that under the umbrella of `CrossBuildMountModcache` and allow the latter to be set to false by an env var

Reply:
+1